src/irmin-pack/s.ml

```diff
@@ -79,6 +79,13 @@ module type S = sig
     ([> `Msg of string ], [> `Msg of string ]) result Lwt.t
 
   val reconstruct_index : ?output:string -> Irmin.config -> unit
+
+  module Stats : sig
+    type paths = step option list list
+    type stat = { length : int * paths; width : int * paths; mp : int * paths }
```

Comment:
Storing the full list here is going to mean millions of entry. 

This could be OK as long as all are not printed at the end of irmin_fsk.

Reply:
I'm only printing an example for each of these three stats, as you suggested - let me know if this is what you had in mind.